cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1353,26 +1353,9 @@ macro(build_snappy)
   set(SNAPPY_CMAKE_ARGS
       ${EP_COMMON_CMAKE_ARGS} -DSNAPPY_BUILD_TESTS=OFF -DSNAPPY_BUILD_BENCHMARKS=OFF
       "-DCMAKE_INSTALL_PREFIX=${SNAPPY_PREFIX}")
-  # Snappy unconditionally enables -Werror when building with clang this can lead
-  # to build failures by way of new compiler warnings. This adds a flag to disable
-  # Werror to the very end of the invocation to override the snappy internal setting.
-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    foreach(CONFIG DEBUG MINSIZEREL RELEASE RELWITHDEBINFO)
-      list(APPEND
-           SNAPPY_CMAKE_ARGS
-           "-DCMAKE_CXX_FLAGS_${UPPERCASE_BUILD_TYPE}=${EP_CXX_FLAGS_${CONFIG}} -Wno-error"
-      )
-    endforeach()
-  endif()
-
-  if(APPLE AND CMAKE_HOST_SYSTEM_VERSION VERSION_LESS 20)
-    # On macOS 10.13 we need to explicitly add <functional> to avoid a missing include error
-    # This can be removed once CRAN no longer checks on macOS 10.13
-    find_program(PATCH patch REQUIRED)
-    set(SNAPPY_PATCH_COMMAND ${PATCH} -p1 -i ${CMAKE_CURRENT_LIST_DIR}/snappy.diff)
-  else()
-    set(SNAPPY_PATCH_COMMAND)
-  endif()
+  # See comments in snappy.diff.
+  find_program(PATCH patch REQUIRED)
+  set(SNAPPY_PATCH_COMMAND ${PATCH} -p1 -i ${CMAKE_CURRENT_LIST_DIR}/snappy.diff)
```

Comment:
@kou Do we require `patch` to be present on all platforms, especially Windows? Or should I make this more specific?

Reply:
Well, to answer myself: our emscripten builds don't have access to a `patch` executable...
https://github.com/ursacomputing/crossbow/actions/runs/10388720694/job/28764995095#step:7:826

Reply:
And even some Linux builds such as `test-r-rocker-r-ver-latest` don't seem to have `patch` either. I'll try to make it optional.

Reply:
Ok, unfortunately the patch is required. We either have to make sure that `patch` is available on all images, or do things differently :-/

Reply:
Yes. `patch` is required as you noticed.
I think that another approach is better. See also: https://github.com/apache/arrow/issues/43688#issuecomment-2289907756

Reply:
OTOH, `patch` is useful and could be needed for other bundled builds.

Reply:
I think that we should fix in upstream instead of using patch as much as possible :-)

Reply:
I agree with this, but as you know Snappy rarely takes outside contributions...